game/views.py

```
@@ -520,4 +520,121 @@ def stats_view(request):
+    if is_best:
+        return 'Best'
+        
+    # If the engine isn't returning a best move (e.g. checkmate), fallback.
+    if not best_mv or not played_mv:
+        return 'Mistake'
+        
+    # Basic heuristic: if the non-best move drops material compared to the best move,
+    # it's a Blunder. Otherwise, a Mistake.
+    piece_vals = {'p': 1, 'n': 3, 'b': 3, 'r': 5, 'q': 9, 'k': 0}
+    
+    def _capture_value(move):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_capture_value`

(ANN202)